src/main/java/apoc/graph/document/builder/DocumentToGraph.java

```diff
@@ -30,21 +33,23 @@ public DocumentToGraph(GraphDatabaseService db, GraphsConfig config) {
         this.config = config;
     }
 
-    public void validate(Map map) {
-        List<String> messages = new ArrayList<>();
-        if (!map.containsKey(config.getIdField())) {
-            messages.add("`" + config.getIdField() + "` as id-field name");
-        }
-        if (!map.containsKey(config.getLabelField())) {
-            messages.add("`" + config.getLabelField() + "` as label-field name");
-        }
-        if (!messages.isEmpty()) {
-            String json = formatDocument(map);
-            throw new RuntimeException("The object `" + json + "` must have " + String.join(" and ", messages));
-        }
+    public Map<Map<String, Object>, List<String>> validate(Map<String, Object> map) {
+        return flatMapFields(map)
+                .map(elem -> {
+                    List<String> msgs = new ArrayList<>();
+                    if (!elem.containsKey(config.getIdField())) {
+                        msgs.add("`" + config.getIdField() + "` as id-field name");
+                    }
+                    if (!elem.containsKey(config.getLabelField())) {
+                        msgs.add("`" + config.getLabelField() + "` as label-field name");
+                    }
+                    return new AbstractMap.SimpleEntry<>(elem, msgs);
+                })
+                .filter(elem -> !elem.getValue().isEmpty())
+                .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
     }
 
-    private String formatDocument(Map map) {
+    public String formatDocument(Map map) {
```

Comment:
make the 100 a parameter and rename the method to indicate that it's only for reporting
like "renderPartialDocument"

Reply:
Can also be done later.